test/core/transport/test_suite/chaotic_good_fixture.cc

```diff
@@ -103,12 +103,36 @@ TRANSPORT_FIXTURE(ChaoticGood) {
           .SetObject(std::static_pointer_cast<EventEngine>(event_engine));
   auto client_transport =
       MakeOrphanable<chaotic_good::ChaoticGoodClientTransport>(
-          std::move(control_endpoints.client), std::move(data_endpoints.client),
+          std::move(control_endpoints.client),
+          chaotic_good::OneDataEndpoint(std::move(data_endpoints.client)),
           ChannelArgs().SetObject(resource_quota), event_engine);
   auto server_transport =
       MakeOrphanable<chaotic_good::ChaoticGoodServerTransport>(
           channel_args, std::move(control_endpoints.server),
-          std::move(data_endpoints.server), event_engine);
+          chaotic_good::OneDataEndpoint(std::move(data_endpoints.server)),
+          event_engine);
+  return ClientAndServerTransportPair{std::move(client_transport),
+                                      std::move(server_transport)};
+}
+
+TRANSPORT_FIXTURE(ChaoticGoodSingleConnection) {
+  auto resource_quota = MakeResourceQuota("test");
+  EndpointPair control_endpoints =
+      CreateEndpointPair(event_engine.get(), resource_quota, 1234);
+  EndpointPair data_endpoints =
```

Comment:
Nit: I dont think we need to create the data_endpoints since they are not used in the single connection test.